### PR TITLE
[stable/grafana] Enabling environment variables to be composed from secrets.

### PR DESCRIPTION
**Pull request by kyschouv**
…riables can be composed from secret values.

Signed-off-by: Kyle Schouviller <kyschouv@microsoft.com>

No

#### What this PR does / why we need it:
Moves env var declarations after envValueFrom declarations, so secrets can be used to compose env var declarations.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/grafana/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: grafana
-version: 4.3.0
+version: 4.3.1
```
Comment (zanhsieh) [inline stable/grafana/Chart.yaml]: Should be 4.3.2 now.

**Comment by zanhsieh** [comment]
/assign

@kyschouv 
Would you mind to re-state your PR subject to description? It gets cutting off and I have hard time to understand what this PR for..

File: stable/grafana/Chart.yaml (hunk 1 above)
Comment (kyschouv) [reply stable/grafana/Chart.yaml]: Done

**Comment by kyschouv** [comment]
@zanhsieh, I updated the title. Basically, if you want to assign environment variables composed from more than one secret value, you need to have the environment variable placed after the `secretKeyRef` variable in the yaml. So by moving the `env` section after `envValueFrom`, it enables this.

In particular, we're trying to use this to bind a MySQL database allocated by OSBA, where the secret has a host and port secret key separately, so I'll define environment variables like:

```
env:
  GF_DATABASE_HOST: "$(GF_DATABASE_HOSTNAME):$(GF_DATABASE_HOSTPORT)"
envValueFrom:
  GF_DATABASE_HOSTNAME:
    secretKeyRef:
      key: host
      name: grafana-database
  GF_DATABASE_HOSTPORT:
    secretKeyRef:
      key: port
      name: grafana-database
```

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**